pennylane/ops/op_math/sum.py

```diff
@@ -0,0 +1,175 @@
+
+    def __repr__(self):
+        """Constructor-call-like representation."""
+        return " + ".join([f"{f}" for f in self.summands])
+
+    def __copy__(self):
+        cls = self.__class__
+        copied_op = cls.__new__(cls)
+        copied_op.data = self.data.copy()  # copies the combined parameters
+        copied_op.summands = tuple(s.__copy__() for s in self.summands)
+
+        for attr, value in vars(self).items():
+            if attr not in {"data", "summands"}:
+                setattr(copied_op, attr, value)
+
+        return copied_op
+
+    @property
+    def num_wires(self):
+        return len(self.wires)
+
+    @property
+    def is_hermitian(self):
+        """If all of the terms in the sum are hermitian, then the Sum is hermitian."""
+        return all([s.is_hermitian for s in self.summands])
+
+    def terms(self):
+        return [1.0]*len(self.summands), self.summands
+
+    @property
+    def eigendecomposition(self):
+        """Return the eigendecomposition of the matrix specified by the Hermitian observable.
+
+        This method uses pre-stored eigenvalues for standard observables where
+        possible and stores the corresponding eigenvectors from the eigendecomposition.
+
+        It transforms the input operator according to the wires specified.
+
+        Returns:
+            dict[str, array]: dictionary containing the eigenvalues and the eigenvectors of the operator
+        """
+        Hmat = self.matrix()
+        Hmat = qml.math.to_numpy(Hmat)
+        Hkey = tuple(Hmat.flatten().tolist())
+        if Hkey not in self._eigs:
+            w, U = np.linalg.eigh(Hmat)
+            self._eigs[Hkey] = {"eigvec": U, "eigval": w}
+
+        return self._eigs[Hkey]
+
+    def diagonalizing_gates(self):
+        """Compute diagonalizing_gates (only if op is hermitian)"""
+
+        eigen_vectors = self.eigendecomposition["eigvec"]
+        return [qml.QubitUnitary(eigen_vectors.conj().T, wires=self.wires)]
+
+    def eigvals(self):
+        """Return the eigenvalues of the specified Hermitian observable.
+
+        This method uses pre-stored eigenvalues for standard observables where
+        possible and stores the corresponding eigenvectors from the eigendecomposition.
+
+        Returns:
+            array: array containing the eigenvalues of the Hermitian observable
+        """
+        return self.eigendecomposition["eigval"]
+
+    def matrix(self, wire_order=None):
+        """Representation of the operator as a matrix in the computational basis."""
+
+        def matrix_gen(summands, wire_order=None):
+            """Helper function to construct a generator of matrices"""
+            for op in summands:
+                yield expand_matrix(op.matrix(), op.wires, wire_order=wire_order)
+
+        if wire_order is None:
+            wire_order = self.wires
+
+        return self._sum(matrix_gen(self.summands, wire_order))
+
+    def _sum(self, mats_gen, dtype=None, cast_like=None):
```

Comment:
I think both `_sum` and `is_commuting` should be moved outside of the class.  Instead of methods attached to `Sum`, they can merely be functions in the `sum.py` submodule.

Reply:
I moved the `_sum` method out of the class def. I don't think we will need `is_commuting` just yet, so I removed it for now, later on we can add it along with `is_unitary`.